nautobot/docs/development/apps/migration/ui-component-framework.md

```diff
@@ -166,27 +178,89 @@ object_detail_content = ObjectDetailContent(
 )
 ```
 
-### 5. Custom Content
+### 6. Custom Content
 
 For custom content that doesn't fit existing panels:
 
 1. Consider using `TextPanel` with custom formatting
+
+```python title="views.py"
+from nautobot.apps import views
+from nautobot.apps.ui import TextPanel
+
+class DeviceDetailView(views.NautobotUIViewSet):
+    panels = [
+        TextPanel(weight=100, context_field="custom_content"),
+    ]
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        context["custom_content"] = "My Custom Content"
+        return context
+```
+
 2. Use `KeyValueTablePanel` with transformed data
+
+```python title="views.py"
+from nautobot.apps import views
+from nautobot.apps.ui import KeyValueTablePanel
+from nautobot.core.templatetags.helpers import (
+    divide,
+    placeholder,
+    slugify,
+    split,
+)
+
+class DeviceDetailView(views.NautobotUIViewSet):
+    panels = [
+        KeyValueTablePanel(weight=100, value_transforms={
+            "name": [slugify, placeholder],
+            "list_of_names": split,
+            "number_value": lambda v: divide(v, 3),
+        }),
+    ]
```

Comment:
change to `DeviceUIViewSet` and add `object_detail_content` as suggested above here as well.